server/room-battle.js

```diff
@@ -783,7 +784,17 @@ class RoomBattle extends RoomGames.RoomGame {
 			parentGame.onBattleWin(this.room, winnerid);
 		}
 		// If the room's replay was hidden, disable users from joining after the game is over
-		if (this.room.hideReplay) this.room.modjoin = '%';
+		if (this.room.hideReplay) {
+			this.room.modjoin = '%';
+			this.room.isPrivate = 'hidden';
+			// We also resave the replay to make sure the hidden setting gets reflected
```

Comment:
I wonder if we should just do this as a general rule – save the replay at the end of a battle if `replaySaved` is true.

Reply:
Done.